### DISCUSSION TITLE
Which Nuget Packages and Stylesheets for Bootstrap5?

### DISCUSSION THREAD

**Discussion by ghost** [Megabit/Blazorise#3375] [Q&A]
Which Nuget Packages do I have to use for Blazorise.Bootstrap 5 (for .NET Core 6 and ServerSide Blazor)?
Only Blazorise.Bootstrap5 or do I need additional Packages?

The problem is that I'm getting some exception:
![grafik](https://user-images.githubusercontent.com/10619439/151675813-29efcd19-1128-41e9-90cf-9921ee6bffee.png)

Do I HAVE to install Blazorise.Icons.FontAwesome? I don't like them and would prefer other icons.

And what Styles do I have to name in _Layouts.cshtml? Currently I'm using
<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <base href="~/" />
    <link href="css/bootstrap/bootstrap.min.css"  rel="stylesheet"/>
    <link href="css/site.css" rel="stylesheet" />
    <link href="MYPROGNAME.styles.css" rel="stylesheet" />
    <component type="typeof(HeadOutlet)" render-mode="ServerPrerendered" />
    ...
a: shall I remove the bootstrap.min.css?
b: do I need some definitions for Bootstrap5 or are they already included in Blazorise.Bootstrap5?

Thanks in advance!

**Comment by ghost** [comment]
Update: I solved the first problem. I overlooked a registration. In Program.cs I had to include:
builder.Services.AddBlazorise(options =>
{
    options.ChangeTextOnKeyPress = true; // optional
});
builder.Services.AddBootstrap5Providers();
builder.Services.AddFontAwesomeIcons();
(AddBlazorise was missing).

Anyway, for the other question I don't have a solution yet. So I'll post them here again:

Do I HAVE to install Blazorise.Icons.FontAwesome? I don't like them and would prefer other icons.

And what Styles do I have to name in _Layouts.cshtml? Currently I'm using
... a: shall I remove the bootstrap.min.css? b: do I need some definitions for Bootstrap5 or are they already included in Blazorise.Bootstrap5?

Thanks in advance!

**Comment by ghost** [comment]
NEXT UPDATE / Next Problem: 

Adding these services seemed to be too much:
![grafik](https://user-images.githubusercontent.com/10619439/151677444-a8b4751b-9e1c-4712-820c-9ab906ecc57f.png)

**Reply by stsrki** [reply]
> NEXT UPDATE / Next Problem:

You've probably added `@using Blazorise.Bootstrap5` in the `_Imports.razor`. Remove it and leave only  `@using Blazorise`.

> Do I HAVE to install Blazorise.Icons.FontAwesome? I don't like them and would prefer other icons.

You can also add `Blazorise.Icons.Material`. Or you can not use the icons package at all. In that case, you would probably need to override `IconProvider` and implement some custom logic because internally Blazorise uses the `Icon` component at some places, and without it it would not work.

**Comment by ghost** [comment]
Yep, that did the trick. Thank you!

As far as I understand now:
The "Blazorise.Bootstrap5" NuGet Package replaces the "Blazorise.Bootstrap" Package, but it's namespace is not "Blazorise.Bootstrap5" (as I have suspected") but didn't change and is still "Blazorise" only. 
And the service registration call for the Bootstrap5-Package is ...AddBlazorise()" too, but ...AddBootstrap5Providers() is an additional call one has to make to get it work.

**Comment by stsrki** [comment]
`Blazorise.Bootstrap5` is not replacing `Blazorise.Bootstrap`. It is just a new provider based on Bootstrap 5. While the existing `Blazorise.Bootstrap` is based on Bootstrap 4. We just didn't want to rename it to `Blazorise.Bootstrap4`. Maybe we can do it at some point.

---

`AddBlazorise()` is used to register necessary services and providers for the core project.
`AddBootstrap5Providers()` is used to register the services so that Blazorise core can pick it up and to know what to do.
